issue_25/get_seq.py

```diff
@@ -0,0 +1,51 @@
+# -*- coding: utf-8 -*-
+"""
+Created on Wed Oct 14 11:25:08 2020
+
+@author: Bátora Dániel
+"""
+
+import os 
+from Bio import SeqIO
+from typing import Union
+
+
+
+def get_seq(filepath:str,organism:Union[str, int],temp_dir:str) -> str:
+    
+    """
+    Selects DNA sequence of the desired organims,
+    outputs it into a FASTA file 
+    
+    Args: 
+        filepath(str): Path to the original FASTA file 
+        organism(Union[str, int]): organism short name(str) or taxon id(int)
+        temp_dir(str): Path to the temporary directory
+        
+    Returns: 
+        output_path(str): Path to the output file saved in the temporary directory
+    
+    """
+    
+    
+    
+    if not os.path.isdir(temp_dir): 
+        os.mkdir(temp_dir)
+    output_path = os.path.join(temp_dir, organism + ".fasta")
+    reads = []
+    with open(filepath,  "r", encoding = "utf-8") as handle:
+        for record in SeqIO.parse(handle, "fasta"):
+            reads.append(record)
+    
+    hits = []
+    for record in reads:
```

Comment:
You are actually going through each record _twice_! Why not check whether you want to keep the record while you are looping over them in line 37?

Reply:
In fact, there is no need to even have a `hits` list because you can just write them out when you loop over them...

Reply:
Can you give me an example on how to do this? Because as it loops over the records, it will only write out the current one and overwrites the previous file. If I store the records in a list, the write function automatically makes the fasta from the list.

Reply:
Check out https://biopython.org/docs/1.75/api/Bio.SeqIO.html
"If you give a filename, then each time you call write() the existing file will be overwritten. For sequential files formats (e.g. fasta, genbank) each “record block” holds a single sequence. For these files it would probably be safe to call write() multiple times by re-using the same handle."